tests/unit/dynamodb/test_conditions.py

```diff
@@ -0,0 +1,453 @@
+        self.value = 'foo'
+        self.value2 = 'foo2'
+
+    def test_and(self):
+        with self.assertRaisesRegexp(
+                DynanmoDBOperationNotSupportedError, 'AND'):
+            self.attr & self.attr2
+
+    def test_or(self):
+        with self.assertRaisesRegexp(
+                DynanmoDBOperationNotSupportedError, 'OR'):
+            self.attr | self.attr2
+
+    def test_not(self):
+        with self.assertRaisesRegexp(
+                DynanmoDBOperationNotSupportedError, 'NOT'):
+            ~self.attr
+
+    def test_eq(self):
+        self.assertEqual(
+            self.attr.eq(self.value), Equals(self.attr, self.value))
+
+    def test_lt(self):
+        self.assertEqual(
+            self.attr.lt(self.value), LessThan(self.attr, self.value))
+
+    def test_lte(self):
+        self.assertEqual(
+            self.attr.lte(self.value), LessThanEquals(self.attr, self.value))
+
+    def test_gt(self):
+        self.assertEqual(
+            self.attr.gt(self.value), GreaterThan(self.attr, self.value))
+
+    def test_gte(self):
+        self.assertEqual(
+            self.attr.gte(self.value),
+            GreaterThanEquals(self.attr, self.value))
+
+    def test_begins_with(self):
+        self.assertEqual(self.attr.begins_with(self.value),
+                         BeginsWith(self.attr, self.value))
+
+    def test_between(self):
+        self.assertEqual(self.attr.between(self.value, self.value2),
+                         Between(self.attr, self.value, self.value2))
+
+
+class TestA(TestK):
+    def setUp(self):
+        self.attr = Attr('mykey')
+        self.attr2 = Attr('myotherkey')
+        self.value = 'foo'
+        self.value2 = 'foo2'
+
+    def test_ne(self):
+        self.assertEqual(self.attr.ne(self.value),
+                         NotEquals(self.attr, self.value))
+
+    def test_is_in(self):
+        self.assertEqual(self.attr.is_in([self.value]),
+                         In(self.attr, [self.value]))
+
+    def test_exists(self):
+        self.assertEqual(self.attr.exists(), AttributeExists(self.attr))
+
+    def test_not_exists(self):
+        self.assertEqual(self.attr.not_exists(), AttributeNotExists(self.attr))
+
+    def test_contains(self):
+        self.assertEqual(self.attr.contains(self.value),
+                         Contains(self.attr, self.value))
+
+    def test_size(self):
+        self.assertEqual(self.attr.size(), Size(self.attr))
+
+    def test_attribute_type(self):
+        self.assertEqual(self.attr.attribute_type(self.value),
+                         AttributeType(self.attr, self.value))
+
+
+class TestConditions(unittest.TestCase):
+    def setUp(self):
+        self.value = Attr('mykey')
+        self.value2 = 'foo'
+
+    def assert_expression_dict(self, condition, exp_format,
+                               exp_operator, values):
+        expression_dict = condition.get_expression()
+        self.assertEqual(expression_dict['format'], exp_format)
```

Comment:
Not sure I follow.  The ordering of key values shouldn't matter since they're unordered.  Can you show an example of this?

```
>>> from nose.tools import assert_dict_equal
>>> assert_dict_equal({'a': 1, 'b': 2, 'c': 3}, {'c': 3, 'b': 2, 'a': 1})
```